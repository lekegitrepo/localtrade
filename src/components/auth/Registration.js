/* eslint-disable camelcase */

import React, { useState } from 'react';
// import PropTypes from 'prop-types';

const Registration = () => {
  const [state, setState] = useState({
    email: '',
    password: '',
    password_confirmation: '',
  });

  const handleForm = e => {
    const userData = '';
    // const state = '';
    console.log('This is initial data', userData, state);
    e.preventDefault();
  };

  const handleChange = e => {
    setState(prev => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <section>
      <h2>Registration Form</h2>
      <form onSubmit={handleForm}>
        <div className="form__group">
          <label htmlFor="fname" className="form__label">
            email
            <input
              type="email"
              name="email"
              className="form__input"
              id="email"
              placeholder="Email"
              value={state.email}
              onChange={handleChange}
              pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$"
              required
            />
          </label>
        </div>
        <div className="form__group">
          <label htmlFor="password" className="form__label">
            Password
            <input
              type="password"
              name="password"
              className="form__input"
              id="password"
              placeholder="Password"
              value={state.password}
              onChange={handleChange}
              required
            />
          </label>
        </div>
        <div className="form__group">
          <label htmlFor="password_confirmation" className="form__label">
            Password Confirmation
            <input
              type="password"
              name="password_confirmation"
              className="form__input"
              id="password_confirmation"
              placeholder="Password Confirmation"
              value={state.password_confirmation}
              onChange={handleChange}
              required
            />
          </label>
        </div>
        <div className="form__group form-btn">
          <button type="submit"> Register </button>
        </div>
      </form>
    </section>
  );
};

/* Registration.defaultProps = {
  loggedIn: false,
};

Registration.propTypes = {
  loggedIn: PropTypes.bool,
}; */

export default Registration;
