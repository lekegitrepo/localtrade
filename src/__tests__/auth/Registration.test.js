import React from 'react';
import { shallow } from 'enzyme';
import Registration from '../../components/auth/Registration';

describe('Registration Component', () => {
  it('should render Registration Component', () => {
    const component = shallow(<Registration />);
    expect(component).toMatchSnapshot();
  })
})
