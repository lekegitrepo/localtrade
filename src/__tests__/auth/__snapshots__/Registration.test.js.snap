// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Component should render Registration Component 1`] = `
<section>
  <h2>
    Registration Form
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <label
        className="form__label"
        htmlFor="fname"
      >
        email
        <input
          className="form__input"
          id="email"
          name="email"
          onChange={[Function]}
          pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,}$"
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </label>
    </div>
    <div
      className="form__group"
    >
      <label
        className="form__label"
        htmlFor="password"
      >
        Password
        <input
          className="form__input"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </label>
    </div>
    <div
      className="form__group"
    >
      <label
        className="form__label"
        htmlFor="password_confirmation"
      >
        Password Confirmation
        <input
          className="form__input"
          id="password"
          name="password_confirmation"
          onChange={[Function]}
          placeholder="Password Confirmation"
          required={true}
          type="password"
          value=""
        />
      </label>
    </div>
    <div
      className="form__group form-btn"
    >
      <button
        type="submit"
      >
         Sign In 
      </button>
    </div>
  </form>
</section>
`;
